clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -88,9 +88,12 @@ public void metricRemoval(KafkaMetric metric) {
         synchronized (LOCK) {
             KafkaMbean mbean = removeAttribute(metric);
             if (mbean != null) {
-                if (mbean.metrics.isEmpty())
+                if (mbean.metrics.isEmpty()) {
                     unregister(mbean);
-                else
+                    MetricName metricName = metric.metricName();
+                    String mBeanName = getMBeanName(prefix, metricName);
```

Comment:
We can compute this once and pass it to `removeAttribute`.

Reply:
To see if I understand your suggestion correctly:
mBeanName is already computed in removeAttribute. Do you mean the check of mbean.metrics.isEmpty() should be done in ```removeAttribute``` (and mbeans.remove() call if that is true) ?

Reply:
No, I mean something like:

```java
String mBeanName = getMBeanName(prefix, metricName);
KafkaMbean mbean = removeAttribute(metric, mBeanName);
...
```